### PR TITLE
Implement dedicated utimeq module for efficient uasyncio scheduling

### PR DESCRIPTION
**Pull request by pfalcon**
This means reverting 0cbc07227c4ab70c846f40a1e2a5fd57dec30428

### CODE REVIEW

File: extmod/modutimeq.c [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+#include <string.h>
+
+#include "py/nlr.h"
+#include "py/objlist.h"
+#include "py/runtime0.h"
+#include "py/runtime.h"
+#include "py/smallint.h"
+
+#if MICROPY_PY_UHEAPQ
```
Comment (dpgeorge) [inline extmod/modutimeq.c]: I don't know if you really meant UTIMEQ here, or if you wanted to keep one config variable (UHEAPQ) that enables both uheapq and utimeq.

File: extmod/modutimeq.c [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    heap->items[heap->len].callback = MP_OBJ_NULL; // so we don't retain a pointer
+    heap->items[heap->len].args = MP_OBJ_NULL;
+    if (heap->len) {
+        heap_siftup(heap, 0);
+    }
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_2(mod_utimeq_heappop_obj, mod_utimeq_heappop);
+
+STATIC mp_obj_t mod_utimeq_dump(mp_obj_t heap_in) {
```
Comment (dpgeorge) [inline extmod/modutimeq.c]: Does this debugging function need to be exposed?  Can it just have a `#if 0` around it?  It makes it hard to write coverage tests for it.

Instead, what about making it return a list of the qentries?

File: extmod/modutimeq.c [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    .name = MP_QSTR_utimeq,
+    .make_new = utimeq_make_new,
+    .unary_op = utimeq_unary_op,
+    .locals_dict = (void*)&utimeq_locals_dict,
+};
+
+STATIC const mp_rom_map_elem_t mp_module_utimeq_globals_table[] = {
+    { MP_ROM_QSTR(MP_QSTR___name__), MP_ROM_QSTR(MP_QSTR_utimeq) },
+    { MP_ROM_QSTR(MP_QSTR_utimeq), MP_ROM_PTR(&utimeq_type) },
+};
```
Comment (dpgeorge) [inline extmod/modutimeq.c]: Could put utimeq object in the uheapq module, to save some bytes.

**Review by dpgeorge — COMMENTED** [review]
Apart from comments it looks ok.  If you think this is the way forward then please merge it.

File: extmod/modutimeq.c (hunk 3 above)
Comment (pfalcon) [inline extmod/modutimeq.c]: I'm not sure, chances that uheapq module will be disabled in some ports with utimeq available.

File: extmod/modutimeq.c (hunk 2 above)
Comment (pfalcon) [inline extmod/modutimeq.c]: Added #if for now.

**Comment by pfalcon** [comment]
Merged.

**Pull request closed without merging**